File: desktop/swift/ChimeraLocale.swift
import Foundation
public enum ChimeraDirection { case ltr, rtl }
public struct ChimeraLocale { public let id: String; public let language: String; public let script: String; public let direction: ChimeraDirection
  public static func from(_ id: String?) -> ChimeraLocale { if id?.lowercased().hasPrefix("ar") == true { return ChimeraLocale(id:"ar-SA",language:"ar",script:"Arabic",direction:.rtl) }; return ChimeraLocale(id:"en-US",language:"en",script:"Latin",direction:.ltr) }
}

File: desktop/java/chimera/i18n/ChimeraLocale.java
package chimera.i18n;
public final class ChimeraLocale {
  public enum Direction { LTR, RTL }
  public final String id, language, script; public final Direction direction;
  public ChimeraLocale(String id, String language, String script, Direction direction) { this.id=id; this.language=language; this.script=script; this.direction=direction; }
  public static ChimeraLocale forId(String id) { return id != null && id.toLowerCase().startsWith("ar") ? new ChimeraLocale("ar-SA","ar","Arabic",Direction.RTL) : new ChimeraLocale("en-US","en","Latin",Direction.LTR); }
}
